cmd/cloud-run/rotate-signify-certificate/rotate_signify_certificate.py

```diff
@@ -0,0 +1,278 @@
+
+        return "Certificate rotated successfully"
+    except ValueError as exc:
+        return prepare_error_response(exc, log_fields)
+
+
+def fetch_new_certificate(
+    cert_data: bytes,
+    private_key: rsa.RSAPrivateKey,
+    access_token: str,
+    certificate_service_url: str,
+):
+    """Fetch new certificates from given certificate service"""
+    old_cert = x509.load_pem_x509_certificate(cert_data)
+
+    csr = (
+        x509.CertificateSigningRequestBuilder()
+        .subject_name(old_cert.subject)
+        .sign(private_key, hashes.SHA256())
+    )
+
+    crt_create_payload = json.dumps(
+        {
+            "csr": {
+                "value": csr.public_bytes(serialization.Encoding.PEM).decode("utf-8")
+            },
+            "validity": {"value": 7, "type": "DAYS"},
+            "policy": "sap-cloud-platform-clients",
+        }
+    )
+
+    cert_create_response = requests.post(
+        certificate_service_url,
+        headers={
+            "Authorization": f"Bearer {access_token}",
+            "Content-Type": "application/json",
+            "Accept": "application/json",
+        },
+        data=crt_create_payload,
+        timeout=10,
+    ).json()
+
+    pkcs7_certs = cert_create_response["certificateChain"]["value"].encode()
+
+    return pkcs7.load_pem_pkcs7_certificates(pkcs7_certs)
+
+
+def prepare_new_secret(
+    certificates: List[x509.Certificate],
+    private_key: rsa.RSAPrivateKey,
+    secret_data: Dict[str, Any],
+    created_at: str,
+) -> Dict[str, Any]:
+    """Prepares new secret data with updated certificates and private key."""
+
+    # format certificates
+    certs_string = ""
+
+    for cert in certificates:
+        certs_string += f"subject={cert.subject.rfc4514_string()}\n"
+        certs_string += f"issuer={cert.issuer.rfc4514_string()}\n"
+        certs_string += f"{cert.public_bytes(Encoding.PEM).decode()}\n"
+
+    private_key_bytes = private_key.private_bytes(
+        Encoding.PEM, PrivateFormat.PKCS8, serialization.NoEncryption()
+    )
+
+    return {
+        "certData": base64.b64encode(certs_string.encode()).decode(),
+        "privateKeyData": base64.b64encode(private_key_bytes).decode(),
+        "createdAt": created_at,
+        "certServiceURL": secret_data["certServiceURL"],
+        "tokenURL": secret_data["tokenURL"],
+        "clientID": secret_data["clientID"],
+        "password": "",  # keep it empty to maintain structure, but indicate it's not needed
+    }
+
+
+def extract_message_data(pubsub_message: Any) -> Any:
+    """Extracts secret rotation message from the Pub/Sub message."""
+    if pubsub_message.get("attributes", {}).get("eventType") != "SECRET_ROTATE":
+        raise ValueError("Unsupported event type")
+
+    data = base64.b64decode(pubsub_message["data"])
+    return json.loads(data)
+
+
+def decrypt_private_key(private_key_data: bytes, password: bytes) -> bytes:
+    """Decrypts an encrypted private key."""
+    # pylint: disable=line-too-long
```

Comment:
please resolve instead of disabling linter...

Reply:
Removed the comment.